src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -148,6 +152,31 @@ public function onKernelController(ControllerEvent $event): void
 
         $request->attributes->set('_mounted_component', $mounted);
 
+        // autowire live file arguments
+        if ($request->files->count()) {
+            $allFiles = $request->files->all();
+            $accessor = PropertyAccess::createPropertyAccessor();
+            foreach (LiveFileArg::liveFileArgs($component, $action) as $parameter => $fileArg) {
+                $path = $fileArg->getPropertyPath();
+
+                if (!$accessor->isReadable($allFiles, $path)) {
+                    throw new \RuntimeException(sprintf('File path "%s" for parameter "%s" is invalid', $fileArg->name, $parameter));
+                }
+
+                if ($files = $accessor->getValue($allFiles, $path)) {
+                    $request->attributes->set(
+                        $parameter,
+                        $fileArg->multiple ? $files : $files[0]
+                    );
+                } else {
+                    $request->attributes->set(
+                        $parameter,
+                        $fileArg->multiple ? [] : null
```

Comment:
If the file was not uploaded, then I think we need to do nothing. Then, naturally, Symfony will (if possible) pass the default value for the argument if there is one (or error if there is not).